client/src/components/Visualizations/DisplayApplications.vue

```diff
@@ -1,7 +1,27 @@
+<script setup lang="ts">
+import { BAlert, BLink } from "bootstrap-vue";
+
+import { DatasetProvider } from "@/components/providers";
+
+interface LinkType {
+    app_name: string;
+    link_name: string;
+}
+
+const props = defineProps({
+    datasetId: {
+        type: String,
+        required: true,
+    },
+});
```

Comment:
```suggestion
interface Props {
    datasetId: string;
}

const props = defineProps<Props>();
```

For consistency with the general style, prefer declaring an interface for props. There will be some advantages to this in Vue3 where you can export from a .vue file and reuse the interfaces in tests, etc.